contrib/scalar/scalar.c

```diff
@@ -695,6 +695,45 @@ static char *remote_default_branch(const char *url)
 	return NULL;
 }
 
+static void strbuf_parentdir(struct strbuf *buf)
+{
+	for (; buf->len > 0 && !is_dir_sep(buf->buf[buf->len]); buf->len--)
+		; /* keep looking for parent directory */
+	buf->buf[buf->len] = '\0';
+}
+
+static int delete_enlistment(void)
+{
+	struct strbuf enlistment = STRBUF_INIT;
+
+	/* Compute the enlistment path (parent of the worktree) */
+	strbuf_addstr(&enlistment, the_repository->worktree);
+	strbuf_parentdir(&enlistment);
+
+#ifdef WIN32
+	/* Change current directory to one outside of the enlistment
+	   so that we may delete everything underneath it. */
+	if (is_inside_dir(enlistment.buf)) {
+		struct strbuf parent = STRBUF_INIT;
+		strbuf_addbuf(&parent, &enlistment);
+		strbuf_parentdir(&parent);
+		if (chdir(parent.buf) < 0)
+			die_errno(_("could not switch to '%s'"),
+				  parent.buf);
+		strbuf_release(&parent);
+	}
+#endif
+
+	if (unregister_dir())
```

Comment:
I missed this earlier, but the regression test didn't (yay tests!): this call to `unregister_dir()` needs to come before we `chdir()` away from the enlistment.

BTW I also realized that we do not even need to ask `is_inside_dir()` on Windows: we know we are, because we called `setup_enlistment_directory()`.

Reply:
> I missed this earlier, but the regression test didn't (yay tests!): this call to `unregister_dir()` needs to come before we `chdir()` away from the enlistment.

Right. I'll have to change this back then. Just to note this is a different behaviour than the C# version.

> BTW I also realized that we do not even need to ask is_inside_dir() on Windows: we know we are, because we called setup_enlistment_directory().

So I can elide the `is_inside_dir(..)` call, but what's the etiquette on declaring that `struct strbuf parent` local variable now that's its outside of a block, but also inside of the `#ifdef`?

Reply:
There are two ways, essentially:
1. enclose the declaration in another `#ifdef` pair
2. enclose the code in an extra `{}` so that you can have a narrower scope for that declaration.

Reply:
Another tack: do that `chdir()` dance also on macOS and Linux ;-)